### ISSUE TITLE
Apple Silicon requires additional autoconfig

### ISSUE THREAD

**Issue by cognominal** [MoarVM/MoarVM#1396]
Currently building on Apple Silicon is broken.
This is a follow up on #1395 that I have accidentally  closed.

The autoconfig should be eventually capable to generate an universal binary.

**Comment by MasterDuke17** [comment]
Closed by https://github.com/MoarVM/MoarVM/pull/1398 correct?

**Comment by cognominal** [comment]
> Closed by #1398 correct?

No, when I opened this issue. I knew doing universal binaries would require additional work which would not be trivial. But I has not clue what it was.

The #1398  conservative patch does only one thing. It does not attempt to JIT on Apple Silicon so the unwary compiling with default options would not be tripped

So as long  one wants  to compile a rakudo without universal binaries that should be fine.
On Apple Silicon, one should even be able to compile rakudo with mac intel binaries with no cross compilation involved, using a shell started using arch -arch i386.

So it should be possible to create a homebrew formula for 10/2020 rakudo star
by incorporating the #1398  patch. Disclaimer: I don't know much about homebrew formulae except discovering you can include a patch.
On Apple Silicon, one can have two homebrew hierarchies, one for each architecture.
The same formula is used for both (I think).
The homebrew team is working on supporting both architecture in one hierarchies.

On the other hand, doing universal binaries is more complex.
Doing universal binaries will permit to distribute one .dmg instead of two.

I see two options. In both case I would like for the user to know nothing but a new
--lipo option. The second seems safer and simpler.

1/  It would involve patching rakudo, nqp and Moarvm repos because all of them generate binaries which are raku-m, nqp-m and moar 
as far as I know.
If I understand correctly the respective  Configure.pl, that will involve segregating most of their code as a &main subroutine, which whill be called once per architecture in different; that is once for arm64 and once for i386.  Then, gather the arch specific binaries to make universal binaries.

2/ doing the same work from a script in rakudo repo
 without patching the Configure.pl except for adding a --lipo option, and only in
the rakudo repo.
The Configure.pl would call the script which would then call again Configure.pl

All this stuff in only plumbing but in place yet unfamiliar to me.
So far, no code. Just discovery work.
